components/nexusoperations/statemachine.go

```diff
@@ -388,13 +388,19 @@ var TransitionCanceled = hsm.NewTransition(
 		enumsspb.NEXUS_OPERATION_STATE_SCHEDULED,
 		enumsspb.NEXUS_OPERATION_STATE_BACKING_OFF,
 		enumsspb.NEXUS_OPERATION_STATE_STARTED,
+		// Requesting cancelation of an unstarted operation transitions the machine to CANCELED and adds a
+		// NEXUS_OPERATION_CANCELED event.
+		// During replication the NEXUS_OPERATION_CANCELED event is applied when the machine is already canceled so we
+		// mark this as a valid transition.
+		enumsspb.NEXUS_OPERATION_STATE_CANCELED,
```

Comment:
does it mean any terminate state (succeeded, failed, timeout) can also transit to canceled? If the SM is canceled before the failover and the SM is timeout after failover, then the cancel event replicated

Reply:
No, this only allows transition from canceled to canceled.